### DISCUSSION TITLE
searching secrets on keywords

### DISCUSSION THREAD

**Discussion by sheepwall** [gopasspw/gopass#2976] [Q&A] [answered]
How would you go about having search match keywords that are not in the filename of a secret?
It would be nice to have as a feature of Gopass to add "tags" or "keywords" to the yaml to enable search. But I understand if that's not desirable or in scope for this project. Another way would be to define another file (containing only secret names and the mentioned search terms) that you search against and then keeping that in sync with the gopass store contents.

Has anybody implemented something for this usecase? I'd be curious for any ideas.

**Comment by buchanan** [comment, marked as answer]
Have you tried the grep sub-command?
```
NAME:
   gopass grep - Search for secrets files containing search-string when decrypted.

USAGE:
   gopass grep command [command options] [needle]

DESCRIPTION:
   This command decrypts all secrets and performs a pattern matching on the content.

COMMANDS:
   help, h  Shows a list of commands or help for one command

OPTIONS:
   --regexp, -r  Interpret pattern as RE2 regular expression (default: false)
   --help, -h    show help
```

**Reply by sheepwall** [reply]
Wow I sure missed that one. Thank you
